rpc/send.go

```diff
@@ -154,121 +143,92 @@ func Send(opts Options, method string, addrs []net.Addr, getArgs func(addr net.A
 	// heartbeat measure ping times. With a bit of seasoning, each
 	// node will be able to order the healthy replicas based on latency.
 
-	replies := []interface{}(nil)
-	helperChan := make(chan interface{}, len(clients))
-	N := opts.N
-	errors := 0
-	retryableErrors := 0
-	successes := 0
-	index := 0
+	done := make(chan *rpc.Call, len(addrs)) // TRIM: maybe do fewer opts.N
 
-	// Send RPCs to replicas as necessary to achieve opts.N successes.
-	for {
-		// Start clients up to N.
-		for ; index < N; index++ {
-			args := getArgs(clients[index].Addr())
-			if args == nil {
-				helperChan <- util.Errorf("nil arguments returned for client %s", clients[index].Addr())
-				continue
-			}
-			reply := getReply()
+	sendFn := func(addr net.Addr) {
+		// NewClient might block while connecting, but connecting in parallel
```

Comment:
Yeah, this comment's placement is weird. It made sense in a previous iteration. Fixing.